packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1278,6 +1359,31 @@ export class GrowthBook<
     return false;
   }
 
+  public getRedirectUrl(): string {
+    return this._redirectedUrl;
+  }
+
+  private _getNavigateFunction():
+    | null
+    | ((url: string) => void | Promise<void>) {
+    if (this._ctx.navigate) {
+      return this._ctx.navigate;
+    } else if (isBrowser) {
+      return (url: string) => {
+        window.location.href = url;
+      };
+    }
+    return null;
+  }
+
+  private _setAntiFlicker() {
+    if (!this._ctx.antiFlicker || !isBrowser) return;
+    const styleTag = document.createElement("style");
+    styleTag.innerHTML = ".gb-anti-flicker { opacity: 0 !important; }";
+    document.head.appendChild(styleTag);
+    document.body.classList.add("gb-anti-flicker");
```

Comment:
We should wrap this whole function body in a try/catch.  Also, many people add growthbook in the `<head>` and if that loads super quickly, `document.body` might be undefined.  Safer to use `document.documentElement` instead, which points to the `<html>` element and will always be defined.